app/javascript/src/LoginPage/loginForms/SignupForm.jsx

```diff
@@ -0,0 +1,127 @@
+// @flow
+
+import React, { Component } from 'react'
+import {
+  Form,
+  ActionGroup
+} from '@patternfly/react-core'
+import {
+  HiddenInputs,
+  FormGroup
+} from 'LoginPage'
+
+type Props = {
+  path: string,
+  user: {
+    email: string,
+    username: string,
+    firstname: string,
+    lastname: string
+  }
+}
+
+type State = {
+  username: string,
+  emailAddress: string,
+  firstname: string,
+  lastname: string,
+  password: string,
+  passwordConfirmation: string,
+  isFilledUsername: ?boolean,
+  isFilledEmailAddress: ?boolean,
+  isFilledFirstname: boolean,
+  isFilledLastname: boolean,
+  isFilledPassword: ?boolean,
+  isFilledPasswordConfirmation: ?boolean
+}
+
+const InputFormGroup = (props) => {
+  const { isRequired, type, value, onChange, isValid } = props
+  const inputProps = {
+    value,
+    onChange,
+    autoFocus: null,
+    inputIsValid: isValid
+  }
+  return (
+    <FormGroup isRequired={isRequired}
+      type={type}
+      labelIsValid={isValid}
+      inputProps={inputProps}
+    />
+  )
+}
+
+class SignupForm extends Component<Props, State> {
+  constructor (props: Props) {
+    super(props)
+    this.state = {
+      username: this.props.user.username,
+      emailAddress: this.props.user.email,
+      firstname: this.props.user.firstname,
+      lastname: this.props.user.lastname,
+      password: '',
+      passwordConfirmation: '',
+      isFilledUsername: undefined,
+      isFilledEmailAddress: undefined,
+      isFilledFirstname: true,
+      isFilledLastname: true,
+      isFilledPassword: undefined,
+      isFilledPasswordConfirmation: undefined
+    }
+  }
+
+  handleTextInputUsername = (username: string) =>
+    this.setState({ username, isFilledUsername: username !== '' })
```

Comment:
@didierofrivia I think we can only check that input fields are not empty, to show the PF4 text hint, and let the browser do the rest of the validation.
I will create a task to add our own client side validation

Reply:
You could just use one fn to validate any field instead of having several and storing the state, maybe on submit

Reply:
The idea is to update the state using ´onChange´, so we can check each field as you type. This is the way it works on the Login Page.
We could also do the validation only on submit. We just need to agree on how to do it